util/encrypt/aes.go

```diff
@@ -120,26 +122,49 @@ func PKCS7Unpad(data []byte, blockSize int) ([]byte, error) {
 
 // AESEncryptWithECB encrypts data using AES with ECB mode.
 func AESEncryptWithECB(str, key []byte) ([]byte, error) {
-	cb, err := aes.NewCipher(key)
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-	blockSize := cb.BlockSize()
-	// The str arguments can be any length, and padding is automatically added to
-	// str so it is a multiple of a block as required by block-based algorithms such as AES.
-	// This padding is automatically removed by the AES_DECRYPT() function.
-	data, err := PKCS7Pad(str, blockSize)
-	if err != nil {
-		return nil, err
-	}
-	crypted := make([]byte, len(data))
-	ecb := newECBEncrypter(cb)
-	ecb.CryptBlocks(crypted, data)
-	return crypted, nil
+	return aesEncrypt(str, key, func(block cipher.Block) cipher.BlockMode {
+		return newECBEncrypter(block)
+	})
 }
 
 // AESDecryptWithECB decrypts data using AES with ECB mode.
 func AESDecryptWithECB(cryptStr, key []byte) ([]byte, error) {
+	return aesDecrypt(cryptStr, key, func(block cipher.Block) cipher.BlockMode {
+		return newECBDecrypter(block)
+	})
+}
+
+// DeriveKeyMySQL derives the encryption key from a password in MySQL algorithm.
+// See https://security.stackexchange.com/questions/4863/mysql-aes-encrypt-key-length.
+func DeriveKeyMySQL(key []byte, blockSize int) []byte {
+	rKey := make([]byte, blockSize)
+	rIdx := 0
```

Comment:
why prefix `Idx` and `Key` with `r`?

Reply:
I have not changed it. It seems that code position change affects 'git diff'.see line#163.